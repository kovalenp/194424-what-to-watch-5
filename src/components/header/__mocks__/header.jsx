import React from "react";

const Header = () => {

  return (
    <header className="page-header user-page__head">
      <p>MockHeader</p>
    </header>
  );
};

export default Header;
