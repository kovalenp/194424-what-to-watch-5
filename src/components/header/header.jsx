import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from 'react-redux';
import { authStatus } from '../../common/constants';

const Header = () => {

  const isAuth = useSelector((state) => state.user.authentication === authStatus.AUTH);
  const avatar = useSelector((state) => state.user.avatar_url);


  let userBlock = (
    <Link
      to={`/login`}
      className="logo__link"
    > Sign In
    </Link>);

  if (isAuth) {
    userBlock = (
      <div className="user-block__avatar">
        <img src={avatar || `img/avatar.jpg`} alt="User avatar" width="63" height="63" />
      </div>
    );
  }

  return (
    <header className="page-header movie-card__head">
      <div className="logo">
        <Link
          to={`/`}
          className="logo__link"
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>
      <div className="user-block">
        { userBlock }
      </div>
    </header>
  );
};

export default Header;
