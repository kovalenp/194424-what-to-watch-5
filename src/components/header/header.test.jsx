import React from "react";
import {MemoryRouter} from 'react-router-dom';
import renderer from "react-test-renderer";
import configureMockStore from 'redux-mock-store';
import {Provider} from "react-redux";

import {Header} from "./header";

const mockStore = configureMockStore();

describe(`Should render Header component correctly`, () => {

  it(`for authenticated user`, () => {
    const store = mockStore({USER: {authentication: `AUTH`}});
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Header />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`for nonauthenticated user`, () => {
    const store = mockStore({USER: {authentication: `NOT_AUTH`}});
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Header />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
