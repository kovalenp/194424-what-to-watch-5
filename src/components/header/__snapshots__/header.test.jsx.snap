// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header component correctly for authenticated user 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`Should render Header component correctly for nonauthenticated user 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="logo__link"
      href="/login"
      onClick={[Function]}
    >
       Sign In
    </a>
  </div>
</header>
`;
